docs/code-generation.md

```diff
@@ -0,0 +1,192 @@
+│       ├── generic.go             # Generic informer access
+│       ├── internalinterfaces/
+│       │   └── factory_interfaces.go
+│       └── tenancy/
+│           ├── interface.go
+│           └── v1alpha1/
+│               ├── interface.go
+│               ├── controlplane.go    # ControlPlane informer
+│               └── postcreatehook.go  # PostCreateHook informer
+└── listers/
+    └── tenancy/
+        └── v1alpha1/
+            ├── controlplane.go        # ControlPlane lister
+            ├── postcreatehook.go      # PostCreateHook lister
+            └── expansion_generated.go # Expansion interfaces
+```
+
+## Usage Examples
+
+### Creating a Clientset
+
+```go
+import (
+    "k8s.io/client-go/rest"
+    clientset "github.com/kubestellar/kubeflex/pkg/generated/clientset/versioned"
+)
+
+func main() {
+    config, err := rest.InClusterConfig()
+    if err != nil {
+        panic(err)
+    }
+
+    client, err := clientset.NewForConfig(config)
+    if err != nil {
+        panic(err)
+    }
+
+    // Use the typed client
+    controlPlanes, err := client.Tenancy().ControlPlanes("").List(ctx, metav1.ListOptions{})
+}
+```
+
+### Using Informers
+
+```go
+import (
+    "time"
+    clientset "github.com/kubestellar/kubeflex/pkg/generated/clientset/versioned"
+    informers "github.com/kubestellar/kubeflex/pkg/generated/informers/externalversions"
+    "k8s.io/client-go/tools/cache"
+)
+
+func main() {
+    client, _ := clientset.NewForConfig(config)
+    
+    // Create shared informer factory
+    factory := informers.NewSharedInformerFactory(client, 30*time.Second)
+    
+    // Get typed informer
+    cpInformer := factory.Tenancy().V1alpha1().ControlPlanes()
+    
+    // Add event handlers
+    cpInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+        AddFunc: func(obj interface{}) {
+            // Handle add
+        },
+        UpdateFunc: func(oldObj, newObj interface{}) {
+            // Handle update
+        },
+        DeleteFunc: func(obj interface{}) {
+            // Handle delete
+        },
+    })
+    
+    // Start the factory
+    stopCh := make(chan struct{})
+    factory.Start(stopCh)
+    
+    // Wait for cache sync
+    factory.WaitForCacheSync(stopCh)
+    
+    // Use the lister (reads from cache, not API server)
+    lister := cpInformer.Lister()
+    cp, err := lister.Get("my-control-plane")
+}
+```
+
+### Using Listers Directly
+
+```go
+import (
+    listers "github.com/kubestellar/kubeflex/pkg/generated/listers/tenancy/v1alpha1"
+    "k8s.io/apimachinery/pkg/labels"
+)
+
+func listControlPlanes(lister listers.ControlPlaneLister) {
+    // List all control planes
+    cps, err := lister.List(labels.Everything())
+    
+    // Get a specific control plane
+    cp, err := lister.Get("my-control-plane")
+}
+```
+
+## When to Regenerate
+
+Regenerate typed clients when:
+
+1. API types in `api/v1alpha1/` are modified
+2. New CRD types are added
```

Comment:
I would combine and extend items 1 and 2 into a single item:

1. Definitions in `api/v1alpha1/` are added, modified, or removed.